### ISSUE TITLE
Drag-and-drop files no longer works

### ISSUE THREAD

**Issue by alexozer** [Kethku/neovide#330]
Previously it was possible to drag-and-drop a file into the neovide window to open it, but it seems like this is no longer possible on 0.5.0. Is this due to SDL2? Will it be re-implemented?

**Comment by Kethku** [comment]
yes this is due to winit. Drag and drop is only implemented for some platforms in sdl2. Eventually we may bring back support for winit in which case it should again work.

**Comment by xarthurx** [comment]
Is there any time line for this drag-n-drop feature? Seems 1.5 yrs has passed.
I'm on Windows platform.

**Comment by MultisampledNight** [comment]
This should be fairly easy to implement, looking at [`winit::event::WindowEvent::DroppedFile`](https://docs.rs/winit/0.24.0/winit/event/enum.WindowEvent.html#variant.DroppedFile). Neovide seems to still have the skeleton of such a drop event:

https://github.com/neovide/neovide/blob/9c1481e1c5b00b4a1f6233665445dcb28cd740be/src/bridge/ui_commands.rs#L144-L145

This could be triggered by matching for ~~winit's~~glutin's event here:

https://github.com/neovide/neovide/blob/9c1481e1c5b00b4a1f6233665445dcb28cd740be/src/window/mod.rs#L142

Is anyone down to writing a few lines? If not, I can do it tomorrow or the day after tomorrow.

**Comment by MultisampledNight** [comment]
Wait, lol. This is already handled:

https://github.com/neovide/neovide/blob/9c1481e1c5b00b4a1f6233665445dcb28cd740be/src/window/mod.rs#L161-L167

**Comment by xarthurx** [comment]
> Wait, lol. This is already handled:
> 
> 
> 
> https://github.com/neovide/neovide/blob/9c1481e1c5b00b4a1f6233665445dcb28cd740be/src/window/mod.rs#L161-L167

Thanks for looking into this.
I don't understand the mechanism, but it seems the drag-n-drop feature is not working in windows...

**Comment by 9mm** [comment]
@fredizzimo  Ok after tons of looking, it seems on OSX the problem is... the plist doesn't contain `CFBundleTypeExtensions`.

I have looked at MacVim's and they have literally about 1000 entries in there, probably for right click context menus (ie, "open with")

I started to research a much simpler list that would work with any file extension for instance, but that's when I realized the second problem which is...

It looks like Cargo bundle doesn't seem to support anything other than the most rudimentary basic list of items :(

I hope I'm misunderstanding that, as it seems just asinine a bundling package doesn't actually let you specify custom values...

I'm not sure if anyone has any insight on that one.

Here are the first few entries of Macvim's as an example:

```xml
<key>CFBundleDocumentTypes</key>
<array>
  <dict>
    <key>CFBundleTypeExtensions</key>
    <array>
      <string>vim</string>
    </array>
    <key>CFBundleTypeIconFile</key>
    <string>MacVim-vim</string>
    <key>CFBundleTypeName</key>
    <string>Vim Script File</string>
    <key>CFBundleTypeRole</key>
    <string>Editor</string>
    <key>LSIsAppleDefaultForType</key>
    <true/>
  </dict>
  <dict>
    <key>CFBundleTypeExtensions</key>
    <array>
      <string>txt</string>
      <string>text</string>
      <string>utf8</string>
    </array>
    <key>CFBundleTypeIconFile</key>
    <string>MacVim-txt</string>
    <key>CFBundleTypeMIMETypes</key>
    <array>
      <string>text/plain</string>
    </array>
    <key>CFBundleTypeName</key>
    <string>Plain Text File</string>
    <key>CFBundleTypeRole</key>
    <string>Editor</string>
    <key>LSIsAppleDefaultForType</key>
    <true/>
  </dict>
  ... 1000 more
```
